channels/rdpdr/client/rdpdr_main.c

```diff
@@ -1652,6 +1652,13 @@ static UINT rdpdr_virtual_channel_event_disconnected(rdpdrPlugin* rdpdr)
 {
 	UINT error;
 
+	/* Remove all dependent devices before tearing down resources */
+	if (rdpdr->devman)
```

Comment:
The goal is to make sure the "Free" for the devices gets called so that there threads will stop running, since they are dependent on the rdpdr channel. I am having trouble reproducing this issue this morning, but when I was having an issue I had devices getting an CHANNEL_NOT_CONNECTED error (it looks like the smartcard device code tries to handles this specific case gracefully). It seemed like it would make sense to try to free these resources earlier, but maybe you are right about the race condition - maybe it should happen right after stopping the rdpdr thread but before disconnecting the channel. I agree this needs to be looked at closer.